bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/camera/CameraHandler.java

```diff
@@ -0,0 +1,144 @@
+ *
+ */
+public class CameraHandler extends NetatmoModuleHandler<NAWelcomeCamera> {
+
+    private static final String LIVE_PICTURE = "/live/snapshot_720.jpg";
+
+    public CameraHandler(@NonNull Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    protected void updateProperties(NAWelcomeCamera moduleData) {
+        updateProperties(null, moduleData.getType());
+    }
+
+    @SuppressWarnings("null")
+    @Override
+    protected State getNAThingProperty(String channelId) {
+        switch (channelId) {
+            case CHANNEL_CAMERA_STATUS:
+                return getStatusState();
+            case CHANNEL_CAMERA_SDSTATUS:
+                return getSdStatusState();
+            case CHANNEL_CAMERA_ALIMSTATUS:
+                return getAlimStatusState();
+            case CHANNEL_CAMERA_ISLOCAL:
+                return getIsLocalState();
+            case CHANNEL_CAMERA_LIVEPICTURE_URL:
+                return getLivePictureURLState();
+            case CHANNEL_CAMERA_LIVEPICTURE:
+                return getLivePictureState();
+            case CHANNEL_CAMERA_LIVESTREAM_URL:
+                return getLiveStreamState();
+        }
+        return super.getNAThingProperty(channelId);
+    }
+
+    protected State getStatusState() {
+        return module != null ? toOnOffType(module.getStatus()) : UnDefType.UNDEF;
+    }
+
+    protected State getSdStatusState() {
+        return module != null ? toOnOffType(module.getSdStatus()) : UnDefType.UNDEF;
+    }
+
+    protected State getAlimStatusState() {
+        return module != null ? toOnOffType(module.getAlimStatus()) : UnDefType.UNDEF;
+    }
+
+    protected State getIsLocalState() {
+        return module != null ? toOnOffType(module.getIsLocal()) : UnDefType.UNDEF;
+    }
+
+    protected State getLivePictureURLState() {
+        return getLivePictureURL() == null ? UnDefType.UNDEF : toStringType(getLivePictureURL());
+    }
+
+    protected State getLivePictureState() {
+        return getLivePictureURL() == null ? UnDefType.UNDEF : HttpUtil.downloadImage(getLivePictureURL());
+    }
+
+    protected State getLiveStreamState() {
+        return getLiveStreamURL() == null ? UnDefType.UNDEF : new StringType(getLiveStreamURL());
+    }
+
+    /**
+     * Get the url for the live snapshot
+     *
+     * @return Url of the live snapshot
+     */
+    private String getLivePictureURL() {
+        String result = getVpnUrl();
+        if (result != null) {
+            result += LIVE_PICTURE;
+        }
+        return result;
+    }
+
+    /**
+     * Get the url for the live stream depending wether local or not
+     *
+     * @return Url of the live stream
+     */
+    private String getLiveStreamURL() {
+        String result = getVpnUrl();
+        if (result != null) {
+            result += "/live/index";
+            result += isLocal() ? "_local" : "";
+            result += ".m3u8";

```

Comment:
You should use a StringBuilder here instead of raw string concatenation.

Reply:
done (but the code wasn't new, it was just moved)